src/coreclr/nativeaot/System.Private.CoreLib/src/System/Threading/Condition.cs

```diff
@@ -107,6 +107,16 @@ public unsafe bool Wait(int millisecondsTimeout)
             Waiter waiter = GetWaiterForCurrentThread();
             AddWaiter(waiter);
 
+            bool isWaitHandleKeywordEnabled = NativeRuntimeEventSource.Log.IsEnabled(
+                EventLevel.Informational,
+                NativeRuntimeEventSource.Keywords.WaitHandleKeyword);
+            if (isWaitHandleKeywordEnabled)
+            {
+                NativeRuntimeEventSource.Log.WaitHandleWaitStart(
+                    NativeRuntimeEventSource.WaitHandleWaitSourceMap.MonitorWait,
+                    0);
```

Comment:
Actually nevermind, the associated object would not be available in the info collected in the profile and it seems like it would be difficult to find the associated object from a heap dump given a Condition object. It seems like it would be easy enough to pass in the associated object from Monitor, that would probably be better. The method's signature could be changed to `bool Wait(int millisecondsTimeout, object? associatedObject = null)` and a pointer to the `associatedObject ?? this` object could be passed in for the event.

Reply:
Are you thinking about creating a new internal method `Condition.Wait`?

Reply:
Could just add a new parameter to the existing method with a default value of null.

Reply:
It seems to be a public API, I'm not sure we want to expose that

Reply:
I'm not sure why Condition is a public class, it seems to be used only from CoreLib and is not in the public API surface ([link](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Runtime/ref/System.Runtime.cs#L15019C11-L15019C11), [link](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Threading/ref/System.Threading.cs)). I expect it would be ok to change the method signature.

Reply:
> I'm not sure why Condition is a public class

Fixed in #95380